testj1939.c

```diff
@@ -105,7 +105,7 @@ int main(int argc, char *argv[])
 		break;
 	case 's':
 		todo_send = strtoul(optarg ?: "8", NULL, 0);
-		if (todo_send > sizeof(dat))
+		if (todo_send > (int)sizeof(dat))
```

Comment:
Hm... strtoul is unsigned long int:
unsigned long int strtoul(const char *nptr, char **endptr, int base);
The sizeof should be size_t, which is unsigned int or unsigned long int, depending on the system.

Why it should be int?

Reply:
todo_send is defined as int.
Is it save to define todo_send as unsigned int ?
sizeof(dat) should never be as high as ULONG_MAX but I was confused about the potential return values given in the man page.

Reply:
If i see it correctly, we will get always unsigned value. On error, we will get ULONG_MAX, in case of strtoul().

Reply:
Ok, then I'll make todo_send an unsigned int.
Thanks!